sns/src/main/scala/akka/stream/alpakka/sns/impl/SnsPublishFlowStage.scala

```diff
@@ -48,7 +48,12 @@ private[sns] final class SnsPublishFlowStage(topicArn: String, snsClient: Amazon
 
       override def onPush(): Unit = {
         isMessageInFlight = true
-        val request = grab(in).withTopicArn(topicArn)
+        val request =
+          topicArn match {
+            case None => grab(in)
```

Comment:
I wonder if it should fail for messages without `topicArn`?
The logic in Alpakka JMS is to have the default destination specified and use that in case a message does not have a per message destination.

Reply:
Don't you think it is kind of confusing? I think we could validate if the PublishRequest/SendMessageRequest does not have a destionation, but, not sure if it'd be a good idea to use just one flow for both features.
Also, I thought we could rename the flows, fixedTopicFlow/fixedTopicSink and fixedQueueFlow/fixedQueueSink are good names for the old flows and sinks.

Reply:
Looking at the existing code, I believe you can just push the setting of the `topicArn` to a `map` in the factory methods so the stages wouldn't know anything about destinations.